libp2p/peer/peerdata.py

```diff
@@ -26,7 +26,9 @@ def set_protocols(self, protocols: Sequence[str]) -> None:
         self.protocols = list(protocols)
 
     def add_addrs(self, addrs: Sequence[Multiaddr]) -> None:
-        self.addrs.extend(addrs)
+        for addr in addrs:
+            if addr not in self.addrs:
+                self.addrs.append(addr)
```

Comment:
What I'm thinking is probably we should change `addrs` from `List[Multiaddr]` to `Set[Multiaddr]` to reduce time complexity. However, this solution works for me so far. 👍

Reply:
0xCf5dE5aEbA9cf8503E6031deAf1279b4f6Db1223